snapcraft/internal/repo/_deb.py

```diff
@@ -186,18 +188,74 @@ def get_packages_for_source_type(cls, source_type):
         return packages
 
     @classmethod
-    def install_build_packages(cls, package_names):
+    def _setup_foreign_arch(cls, arch):
+        native_arch = subprocess.check_output(
+            ['dpkg', '--print-architecture']).decode(
+                sys.getfilesystemencoding())
+        if arch in native_arch:
+            return
+        foreign_archs = subprocess.check_output(
+            ['dpkg', '--print-foreign-architectures']).decode(
+                sys.getfilesystemencoding())
+        if arch in foreign_archs:
+            return
+        logger.info('Adding foreign architecture {}'.format(arch))
+        subprocess.check_output(
+            ['sudo', 'dpkg', '--add-architecture', arch])
```

Comment:
I know I've said this before, but I'll add my voice here again: I personally don't like this. It means running the wrong snapcraft.yaml on my host will garbage up my arch and sources. The only place this makes sense if when we're building in containers by default, but we're not there yet. Is there any way we can just warn if we're not building in a container?

Reply:
Warn and then what? The build will fail.

Reply:
Yes, that's exactly what I want.

Reply:
Ah, yeah. I see what you're saying now. We could basically generate the necessary commands and leave it up to you to decide which way to go - to my mind, that's the crucial bit, make it as discoverable as possible what's needed to continue, since it's just tedious to find this out for yourself if you've never done it.

How about something like `The host needs to be setup for the target architecture. You can use cleanbuild or set SNAPCRAFT_CONTAINER_BUILDS=1 to build in a container that's automatically set up for cross-compiling. To setup the local machine, run the following commands: [...]`.
Long term with #1458 we could also consider a prompt like `Building in a container is recommended for cross-compiling [...] [y/n]`

Reply:
Yeah that's a great error, +1.